server/src/instant/model/rule.clj

```diff
@@ -147,7 +147,7 @@
     (let [compiler (cel/action->compiler action)]
       (if (and (= "$users" etype)
                (#{"view" "link" "unlink"} action))
-        (let [code "auth.id == data.id"
+        (let [code "auth.id == data.id || (data.linkedPrimaryUser != null && auth.id == data.linkedPrimaryUser)"
```

Comment:
This is the rule change that allows users to see the list of guest users that are linked to them.

Users can't modify the data in the system tables (except for $files.path), so we don't have to worry about them linking to users they shouldn't. When we open up that ability, we should just prevent users from editing values for any of the system attributes (excepting $files.path).